### PR TITLE
template: add additional variable for dest path

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Dustin Spicuzza <dustin@virtualroadside.com>
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/files/template.py
lib/ansible/modules/windows/win_template.py
lib/ansible/plugins/action/template.py
lib/ansible/template/__init__.py

### CODE REVIEW

File: lib/ansible/template/__init__.py [hunk 1]
```diff
@@ -84,20 +84,24 @@
...
+        'template_host': to_text(os.uname()[1]),
+        'template_path': path,
+        'template_mtime': datetime.datetime.fromtimestamp(os.path.getmtime(b_path)),
+        'template_uid': to_text(template_uid),
+        'template_fullpath': os.path.abspath(path),
+        'template_run_date': datetime.datetime.now(),
+    }
+
+    if dest_path:
+        temp_vars['template_destpath'] = to_native(dest_path)
```
Comment (dagwieers) [inline lib/ansible/template/__init__.py]: I think it's best to always provide **template_destpath**, and if it wasn't defined return `jinja.StrictUndefined` or possible `None`.

**Review by dagwieers — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@dagwieers Could you please take another look ? Thanks.

**Review by gundalow — APPROVED** [review]
Working and formatting looks good
Thanks for removing the "The six additional" bits

**Comment by Akasurde** [comment]
@dagwieers Could you please take another look into this ? Thanks.

**Review by dagwieers — APPROVED** [review]

**Comment by Akasurde** [comment]
@gundalow @dagwieers Thanks for the reviews.

**Pull request merged**